docs/1161_SPECS.md

```diff
@@ -0,0 +1,58 @@
+### Specification
+
+- Each component will have a uniqueID
+- uniqueID should be formed by componentName. 
+- Settings store will be enhanced with a new key `feedback`
+- each entry in feedback will be indexed by componentID, and containing the following fields
+
+#### Data model
+```
+settings: {
+	feedback: {
+		enabledIn: [[String], [String]],
+		showInterval: [Date/Millisecconds],
+		components: {
+			[componentID]: {
+				lastGivenOn: [Date/Millisecconds],
+				choosenToBeHidden: [Boolean],
+			}
+		}
+	}
+}
+```
+
+#### Implementation details
+Pseudo code for rendering a component:
+
+```
+onFeedbackShouldShow() =>
+
+if settings.feedback.enabledIn[componentId]
+
+  componentObj  = settings.feedback.components.componentId
+  todayDate = new Date().getTime()
+
+  if !componentObj || !componentObj.choosenToBeHIdden
+      if componentObj.lastGivenOn > todayDate + settings.feedback.showInterval
+      componentObj.lastGivenOn = todayDate;
+      render feature feedback;
+      return;
+
+render component;
+...
+on feedback closed
+if user choosenToBeHIdden 
+	componentObj.choosenToBeHIdden = true
```

Comment:
Little bit of nit-picking: inconsistent capitalization of `choosenToBeHIdden` should be `choosenToBeHidden` here and in few other places.

Reply:
updated specs